model/src/payout_curve/quanto.rs

```diff
@@ -231,45 +230,15 @@ impl Curve {
     /// Price interval at which the party going long gets liquidated i.e. their payout amount equals
     /// zero.
     fn long_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
-        let initial_margin = self
-            .initial_margin_long()
-            .to_signed()
-            .context("Could not convert long's initial margin to bitcoin::SignedAmount")?;
-        let effective_initial_margin =
-            initial_margin + self.fee_offset.as_signed_amount(Position::Long);
-        let effective_initial_margin = effective_initial_margin
-            .to_unsigned()
-            .context("Could not convert long's effective initial margin to bitcoin::Amount")?;
-
-        let bankruptcy_price = bankruptcy_price_long(
-            effective_initial_margin,
-            self.n_contracts,
-            self.initial_price,
-            self.multiplier,
-        );
+        let bankruptcy_price = bankruptcy_price_long(self.initial_price, self.leverage_long);
```

Comment:
🤔 I think this will lead to the liquidation price not changing as we roll over and accumulate fees. 

As I mentioned in private, I think this works for the offer because there are minimal (perhaps zero) fees at that stage and it's just an informative figure.

But as we roll over I think we need to take into account how the effective margins change. Otherwise the liquidation price will remain the same. Maybe that's what we want, but that's not how we implemented this at first, so we at least need to think about this change.

Reply:
As far as I can remember this is an old problem that boils down to the margin changing because the interest (i.e. fees) is deducted from the funds locked up (which is a constraint that cannot easily be changed in our solution). The liquidation price calculated changing with the effective margin is not a feature we actively perused so far but I suppose it would be mathematically more correct. I wonder if this is something that a user would expect in behavior compared to other trading solutions out there.

Reply:
My intuition would be that it's more correct. It would be interesting to consider how this behaviour compares to the inverse curve as well.

I don't know what's best, but if we decide to change it, could you document this change and the implications in the commit message, @bonomat?

Reply:
I think it just clicked, let me see if I can explain it. 

The formula I used is correct but @luckysori is right, we will need to take the margin into account when doing rollover as the new bankruptcy price will differ.

The initial margin was calculated with 
`margin=(n_contracts * initial_price * Multiplier) / leverage`

but over time, the margin is getting reduced (or increased) through the fee account. 

What we have to do is calculate the _effective leverage_, i.e. the inverse of the above formula: 

`effective_leverage=(n_contracts * initial_price * Multiplier) / margin`

So basically, if we take my simplified formula of calculating the bankruptcy price of

`=EntryPrice - (EntryPrice * ((1/leverage) + InitialMargin))` 

we would need to swap `leverage` with the `effective leverage` calculated using the reduced (or increased) margin when rolling over and should get an accurate bankruptcy price when rolling over.

Does this make sense? I've played around here: 

https://docs.google.com/spreadsheets/d/1Hm9BKz1EKymGpNyto0SNShQ-CdZPC9EJqWwklvEtQ-w/edit#gid=0


I think that's actually what @luckysori was doing with his formulas but I did not get it.

Reply:
The same applies for inverse contracts as well.

Reply:
Implemented in [`f2599c1` (#2772)](https://github.com/itchysats/itchysats/pull/2772/commits/f2599c163bf6761aeca49e84953692625ffe7575)